### PR TITLE
Add more information about how `mike` works

### PR DESCRIPTION
**Pull request by llucax**
When generating GitHub pages, it is not obvious how mike works and how to manage multiple versions and aliases. This improves the documentation to make it a bit easier to understand.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -149,6 +149,37 @@ Then make sure that GitHub Pages is enabled in
...
+via GitHub Actions on any push.
+
+If a push is to a branch with semver-like format (vX.Y.Z), then the generated
+version will replace the current `next` version.
+
+If a tag is pushed instead, then the generated version will replace the current
+`vX.Y` version (if there was any), and the aliases `vX` and `latest` will be
+updated to point to the new version.
+
+So basically, we don't create versions for patch releases, only for minor
```
Comment (Marenz) [inline docs/index.md]:
```suggestion
To summarize, we don't create versions for patch releases, only for minor
```

**Review by Marenz — DISMISSED** [review]
Made a suggestion, but feel free to merge anyway

**Comment by daniel-zullo-frequenz** [comment]
I've removed the PR from the queue in case you'd like to address the suggestion

**Comment by llucax** [comment]
Thanks @daniel-zullo-frequenz! Done and re-added to the queue, but now it needs a new approval.

File: docs/index.md [hunk 2]
```diff
@@ -149,6 +149,37 @@ Then make sure that GitHub Pages is enabled in
...
+[Mike](https://pypi.org/project/mike/), which is used to keep multiple versions
+of the website.
+
+The new documentation version is called `next`, which is used as the name for
+the currently in-development branch. The `next` branch has an alias called
+`latest`, which is set as the *default*.
+
+If the website is visited without specifying an explicit version, the `latest`
+version will be displayed. It is recommended to point `latest` to the latest
+stable version instead of the currently in-devepoment version, so as you as you
```
Comment (daniel-zullo-frequenz) [inline docs/index.md]: duplicated as you as you

Sorry I've just spotted it

**Review by daniel-zullo-frequenz — COMMENTED** [review]
a cosmetic comment and LGTM

**Review by matthias-wende-frequenz — COMMENTED** [review]
lgtm too

File: docs/index.md (hunk 2 above)
Comment (llucax) [reply docs/index.md]: Actually `s/you/soon/`: "as soon as you" :)

Comment (llucax) [reply docs/index.md]: Fixed.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**